### ISSUE TITLE
google.sent_time expected String -> crash

### ISSUE THREAD

**Issue by m1ga** [morinel/gcmpush#45]
Not sure if something changed at googles side because I had that working a week ago and now I receive this message:
```
D/GCMIntentService(5503): (IntentService[GCMIntentService--2]) [8312,16380] Push notification received
W/Bundle(5503): Key google.sent_time expected String but value was a java.lang.Long.  The default value <null> was returned.
W/Bundle(5503): Attempt to cast generated internal exception:
W/Bundle(5503): java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.String
W/Bundle(5503): 	at android.os.BaseBundle.getString(BaseBundle.java:923)
W/Bundle(5503): 	at nl.vanvianen.android.gcm.GCMIntentService.onMessage(GCMIntentService.java:98)
W/Bundle(5503): 	at com.google.android.gcm.GCMBaseIntentService.onHandleIntent(GCMBaseIntentService.java:223)
W/Bundle(5503): 	at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:66)
W/Bundle(5503): 	at android.os.Handler.dispatchMessage(Handler.java:102)
W/Bundle(5503): 	at android.os.Looper.loop(Looper.java:168)
W/Bundle(5503): 	at android.os.HandlerThread.run(HandlerThread.java:61)
D/GCMIntentService(5503): (IntentService[GCMIntentService--2]) [1,16381] Message key: "google.sent_time" value: "null"
W/Bundle(5503): Key google.sent_time expected String but value was a java.lang.Long.  The default value <null> was returned.
W/Bundle(5503): Attempt to cast generated internal exception:
W/Bundle(5503): java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.String
W/Bundle(5503): 	at android.os.BaseBundle.getString(BaseBundle.java:923)
W/Bundle(5503): 	at nl.vanvianen.android.gcm.GCMIntentService.onMessage(GCMIntentService.java:106)
W/Bundle(5503): 	at com.google.android.gcm.GCMBaseIntentService.onHandleIntent(GCMBaseIntentService.java:223)
W/Bundle(5503): 	at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:66)
W/Bundle(5503): 	at android.os.Handler.dispatchMessage(Handler.java:102)
W/Bundle(5503): 	at android.os.Looper.loop(Looper.java:168)
W/Bundle(5503): 	at android.os.HandlerThread.run(HandlerThread.java:61)
E/AndroidRuntime(5503): FATAL EXCEPTION: IntentService[GCMIntentService--2]
E/AndroidRuntime(5503): java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.startsWith(java.lang.String)' on a null object reference
E/AndroidRuntime(5503): 	at nl.vanvianen.android.gcm.GCMIntentService.onMessage(GCMIntentService.java:108)
E/AndroidRuntime(5503): 	at com.google.android.gcm.GCMBaseIntentService.onHandleIntent(GCMBaseIntentService.java:223)
E/AndroidRuntime(5503): 	at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:66)
E/AndroidRuntime(5503): 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime(5503): 	at android.os.Looper.loop(Looper.java:168)
E/AndroidRuntime(5503): 	at android.os.HandlerThread.run(HandlerThread.java:61)
```

It looks like this section:
https://github.com/morinel/gcmpush/blob/master/src/nl/vanvianen/android/gcm/GCMIntentService.java#L108
has to include a null check for value

**Comment by RavindraChherke** [comment]
Crashing in my app too. Push works on Nexus 5 but crashing on other devices (Lava android one).

**Comment by m1ga** [comment]
My device is a HTC A9 (Andorid 6) and it used to work there too. I even recorded a video two weeks ago showing the push feature :smile:

**Comment by DFoxinator** [comment]
@m1ga 

I'm having a very weird issue surrounding this area of the code too and it seems too close to be coincidental. All of a sudden (starting a few days ago), my app has started crashing a lot with:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.startsWith(java.lang.String)' on a null object reference
	at nl.vanvianen.android.gcm.GCMIntentService.onMessage(GCMIntentService.java:105)
```

We didn't make any changes to our push notif implementation and although we released a new build a few days prior to the start of the error happening a lot, we didn't change anything with push notifs.

It makes me think something had to change on the Google side or Android or something...

Does anyone have any insight into this?

**Comment by DFoxinator** [comment]
I'm also seeing this same error in addition to the one I posted above. I'm really not sure what happened here.

**Comment by DFoxinator** [comment]
This PR fixes the issue: https://github.com/morinel/gcmpush/pull/46

But still not clear why it started happening.

**Comment by skypanther** [comment]
Sorry, the PR to this repo was a mistake. I meant only to merge the fix posted here to my own fork of the repo. For those of you not able to build the module from source, you'll find a compiled version in the dist folder of my fork with the version number set to 1.5 (yeah, I know that's behind this repo's version number)

**Comment by morinel** [comment]
Fixed in release 1.7

**Comment by AppWerft** [comment]
my observation: this happens outside the module code. If I outcomment the implamentation  of `onMessageReceived` the error comes also, but the message will shown in tray. I sent:
```
{
	"to": "eR4VQf0t2Ek:APA91bESvrI1QFd9CI8wQJvwGgASBZeiZGIFK6FPwXE_OdMilRvUUmjn0yP3k5bo3sXoniWetM7esH1knJHz6Hx52ldxF0l6G3DnCzdHgF76G0PMpB6MfYOR33PYyWAOpBNCX14tZN0Z",
	"time_to_live": 108,
	"delay_while_idle": true,
	"notification": {
		"title": "Title2",
		"body" : "Hello1843",
		"score": "4x8",
		"time": "15:16.2342"
	},
	"description": "Simple push for demonstrating"
}
```
with postman
